unity-renderer/Assets/Features/ScreenshotCamera/Scripts/ScreenshotCapture.cs

```diff
@@ -0,0 +1,72 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI.InWorldCamera.Scripts
+{
+    public class ScreenshotCapture
+    {
+        private const int DESIRED_WIDTH = 1920;
+        private const int DESIRED_HEIGHT = 1080;
+
+        private readonly Camera screenshotCamera;
+        private readonly Sprite sprite;
+        private readonly RectTransform imageRectTransform;
+        private readonly RectTransform canvasRectTransform;
+
+        private (float width, float height) spriteRect;
+
+        public ScreenshotCapture(Camera screenshotCamera, RectTransform canvasRectTransform, Image refBoundariesImage)
+        {
+            sprite = refBoundariesImage.sprite;
+            imageRectTransform = refBoundariesImage.rectTransform;
+
+            this.screenshotCamera = screenshotCamera;
+            this.canvasRectTransform = canvasRectTransform;
+        }
+
+        public byte[] CaptureScreenshot()
+        {
+            spriteRect = GetCurrentSpriteResolution(imageRectTransform.rect, sprite.bounds);
+
+            float scaleFactorW = DESIRED_WIDTH / spriteRect.width;
+            float scaleFactorH = DESIRED_HEIGHT / spriteRect.height;
+
+            int renderTextureWidth = Mathf.RoundToInt(canvasRectTransform.rect.width * scaleFactorW);
+            int renderTextureHeight = Mathf.RoundToInt(canvasRectTransform.rect.height * scaleFactorH);
+
+            var renderTexture = RenderTexture.GetTemporary(renderTextureWidth, renderTextureHeight, 24);
+
+            screenshotCamera.targetTexture = renderTexture;
+            screenshotCamera.Render();
+            RenderTexture.active = screenshotCamera.targetTexture;
+
+            float cornerX = (renderTextureWidth - DESIRED_WIDTH) / 2f;
+            float cornerY = (renderTextureHeight - DESIRED_HEIGHT) / 2f;
+
+            var texture = new Texture2D(Mathf.RoundToInt(DESIRED_WIDTH), Mathf.RoundToInt(DESIRED_HEIGHT), TextureFormat.RGB24, false);
+            texture.ReadPixels(new Rect(cornerX, cornerY, DESIRED_WIDTH, DESIRED_HEIGHT), 0, 0);
+            texture.Apply();
+
+            // Clean up
+            screenshotCamera.targetTexture = null;
+            RenderTexture.ReleaseTemporary(renderTexture);
+
+            return texture.EncodeToJPG();
```

Comment:
Why not PNG?

Reply:
10 times smaller (so will be cheaper on the server side). Anyway, we don't have alpha

Reply:
Makes sense!